### PR TITLE
Add periodic workflow for installing private packages

### PR DESCRIPTION
**Pull request by hasheddan**
Adds a periodic workflow for testing installing private packages with
latest Crossplane build.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### CODE REVIEW

File: .github/workflows/private-package.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: azure/setup-kubectl@v1
+      - name: Create Namespace
+        run: kubectl create namespace crossplane-system
+        shell: bash
+      # https://cloud.google.com/artifact-registry/docs/access-control#pullsecrets
+      - name: Create Namespace
+        run: |
+          kubectl create secret docker-registry package-pull-secret -n crossplane-system \
+          --docker-server=us-west1-docker.pkg.dev \
+          --docker-email=${{ secrets.GAR_EMAIL }} \
```
Comment (hasheddan) [inline .github/workflows/private-package.yml:39]: @jbw976 can you add the `GAR_EMAIL` secret to this repo, which should match the email in the `GAR_SA_KEY`?

File: .github/workflows/private-package.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - cron: '0 12 * * *'
+  pull_request: {}
+  workflow_dispatch: {}
+
+env:
+  GO_VERSION: '1.18'
+  KIND_VERSION: 'v0.14.0'
+
+jobs:
+  provider-upgrade-crossplane-stable:
```
Comment (jbw976) [inline .github/workflows/private-package.yml]: can we make this a more applicable name? e.g. `private-package-pull`?

File: .github/workflows/private-package.yml (hunk 1 above)
Comment (jbw976) [reply .github/workflows/private-package.yml:39]: yes sir, i will do that right now!!

File: .github/workflows/private-package.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          --docker-server=us-west1-docker.pkg.dev \
+          --docker-email=${{ secrets.GAR_EMAIL }} \
+          --docker-username=${{ secrets.GAR_USER }} \
+          --docker-password=${{ secrets.GAR_SA_KEY }}
+        shell: bash
+      - name: Install Latest Crossplane
+        run: helm repo add crossplane-master https://charts.crossplane.io/master/ && helm repo update && helm install crossplane --namespace crossplane-system crossplane-master/crossplane --devel --wait
+        shell: bash
+      - name: Run Private Package Install Tests
+        run: go test -timeout 10m -v --tags=e2e_package ./test/...
```
Comment (jbw976) [inline .github/workflows/private-package.yml:47]: @parul5sahoo take special note of this tag that's being used - your test file would also need to include it as a build tag at the top of the file.  similar to what we see in the provider upgrade tests in https://github.com/crossplane/test/blob/master/test/provider/upgrade/upgrade_test.go#L1

**Review by jbw976 — APPROVED** [review]
Just a small comment about the name of the test being used :)

File: .github/workflows/private-package.yml (hunk 2 above)
Comment (hasheddan) [reply .github/workflows/private-package.yml]: whoops! Yes absolutely :)

File: .github/workflows/private-package.yml (hunk 1 above)
Comment (jbw976) [reply .github/workflows/private-package.yml:39]: done! i also included it in vault 🔒

File: .github/workflows/private-package.yml (hunk 3 above)
Comment (parul5sahoo) [reply .github/workflows/private-package.yml:47]: ohkay got it! And should I go ahead and add in the copyright and license stuff?

**Review by hasheddan — COMMENTED** [review]
I believe we have to merge before this new workflow can access secrets, so I'll move ahead and circle back if it is still having trouble 👍🏻

**Pull request merged**